### PR TITLE
Auto-label PRs based on their content [skip-ci]

### PR DESCRIPTION
**Pull request by jolorunyomi**
This PR adds the GitHub action [PR Labeler](https://github.com/actions/labeler) to auto-label PRs based on their content. 

Labeling is managed with a configuration file `.github/labeler.yml` using the following [options](https://github.com/actions/labeler#usage).

### CODE REVIEW

**Comment by jolorunyomi** [comment]
Labeller config was added in error in https://github.com/rapidsai/cuxfilter/commit/99d7cf67802270d24db0051162df4feb798f2e15

**Review by ajschmidt8 — APPROVED** [review]

File: .github/labeler.yml [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
- - conda/**
- - **/Dockerfile
- - **/.dockerignore
- - docker/**
+ - '.github/**'
+ - 'ci/**'
+ - 'conda/**'
+ - '**/Dockerfile'
+ - '**/.dockerignore'
+ - 'docker/**'
```
Comment (mike-wendt) [inline .github/labeler.yml:25]: This section should be removed

File: .github/labeler.yml [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
 
 CMake:
- - **/CMakeLists.txt
- - **/cmake/**
+ - '**/CMakeLists.txt'
+ - '**/cmake/**'
 
 Java:
- - java/**
+ - 'java/**'
```
Comment (mike-wendt) [inline .github/labeler.yml:17]: This section should match this repo's labels and the contents of the .github/CODEOWNERS file as a starting guide - PICs in this repo would then add or suggest more labels as needed

**Review by mike-wendt — CHANGES_REQUESTED** [review]

**Review by ajschmidt8 — CHANGES_REQUESTED** [review]
The `Python`, `cpp`, and `CMake` labels don't yet exist in this repo. Check with Mike on how to handle missing labels.

**Review by mike-wendt — APPROVED** [review]

**Comment by mike-wendt** [comment]
@ajschmidt8 can you re-review?
@exactlyallan can you review for this repo? please suggest any changes or additions

**Review by AjayThorve — APPROVED** [review]

**Comment by AjayThorve** [comment]
@mike-wendt can we add auto-label for `docs` folder as well?

cc @exactlyallan

**Review by ajschmidt8 — APPROVED** [review]

File: .github/labeler.yml [hunk 3]
```diff
@@ -1,25 +1,13 @@
...
-
-Ops:
- - .github/**
- - /ci/**
- - conda/**
- - **/Dockerfile
- - **/.dockerignore
- - docker/**
+conda:
+   - 'conda/**'
```
Comment (mike-wendt) [inline .github/labeler.yml:13]:
```suggestion
   - 'conda/**'
 
 doc:
   - 'docs/**'
```

Comment (mike-wendt) [reply .github/labeler.yml:13]: This is to address the request from @AjayThorve and @exactlyallan

File: .github/labeler.yml [hunk 4]
```diff
@@ -1,25 +1,16 @@
...
+ 
+ doc:
```
Comment (mike-wendt) [inline .github/labeler.yml:15]:
```suggestion
doc:
```
Fixing indentation issue

**Review by mike-wendt — APPROVED** [review]

**Comment by mike-wendt** [comment]
> @mike-wendt can we add auto-label for `docs` folder as well?
> 
> cc @exactlyallan

Done

**Comment by mike-wendt** [comment]
@exactlyallan feel free to merge when ready with the bot

**Pull request merged**